core/win32/callback.c

```diff
@@ -3115,6 +3115,18 @@ intercept_new_thread(CONTEXT *cxt)
     is_client = is_new_thread_client_thread(cxt, &dstack);
     if (is_client) {
         ASSERT(is_dynamo_address(dstack));
+        /* i#2335: we support setup separate from start, and we want to allow a client
+         * to create a client thread during init, but we do not support that thread
+         * executing until the app has started (b/c we have no signal handlers in place).
+         */
+        /* i#3973: on unix, there are some race conditions that can
+         * occur if the client thread starts executing before
+         * dynamo_initialized is set.  although we are not aware of
```

Comment:
style nit: Capitalize complete sentences in comments (i.e., s/although/Although/) (optionally s/on unix/On unix/ too; also note that some old comments in the code do not currently follow this as the style rule was adopted later)